vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/test/java/com/vaadin/flow/component/sidenav/tests/SideNavItemTest.java

```diff
@@ -437,47 +441,236 @@ public void createFromComponent_setQueryParameters_updateQueryParameters_pathIsU
+        runWithMockRouter(() -> {
             SideNavItem item = new SideNavItem("test",
                     TestRouteWithRouteParams.class,
                     new RouteParameters(Map.of("k1", "v1", "k2", "v2")));
 
             Assert.assertEquals("foo/v1/v2/bar", item.getPath());
-        });
+        }, TestRouteWithRouteParams.class);
     }
 
     @Test
-    public void setPathFromComponentAndRouteParameters_pathContainsParameters() {
-        runWithMockRouter(TestRouteWithRouteParams.class, () -> {
+    public void setPathAndRouteParametersAsComponent_pathContainsParameters() {
+        runWithMockRouter(() -> {
             SideNavItem item = new SideNavItem("test");
             item.setPath(TestRouteWithRouteParams.class,
                     new RouteParameters(Map.of("k1", "v1", "k2", "v2")));
 
             Assert.assertEquals("foo/v1/v2/bar", item.getPath());
-        });
+        }, TestRouteWithRouteParams.class);
     }
 
-    private void runWithMockRouter(Class<? extends Component> route,
-            Runnable test) {
-        Router router = mockRouter(route);
+    @Test
+    public void createFromComponentWithRouteParameters_aliasContainsParameters() {
+        runWithMockRouter(() -> {
+            SideNavItem item = new SideNavItem("test",
+                    TestRouteWithAliases.class,
+                    new RouteParameters(Map.of("key", "value")));
+
+            Assert.assertTrue(item.getPathAliases().contains("foo/value/bar"));
+        }, TestRouteWithAliases.class);
+    }
+
+    @Test
+    public void setPathAndRouteParametersAsComponent_aliasContainsParameters() {
+        runWithMockRouter(() -> {
+            SideNavItem item = new SideNavItem("test");
+            item.setPath(TestRouteWithAliases.class,
+                    new RouteParameters(Map.of("key", "value")));
+
+            Assert.assertTrue(item.getPathAliases().contains("foo/value/bar"));
+        }, TestRouteWithAliases.class);
+    }
+
+    @Test
+    public void createWithNoPathAlias_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setEmptyStringAsPathAlias_pathAliasAdded() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases(Set.of(""));
+
+        Assert.assertEquals(Set.of(""), item.getPathAliases());
+        Assert.assertEquals("[\"\"]",
+                item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setMultiplePathAliases_pathAliasesAdded() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases(Set.of("alias1", "alias2"));
+
+        Assert.assertEquals(Set.of("alias1", "alias2"), item.getPathAliases());
+        Assert.assertTrue(Set
+                .of("[\"alias1\",\"alias2\"]", "[\"alias2\",\"alias1\"]")
```

Comment:
Looks like the order of aliases is unstable. Why?

Reply:
We currently handle them as sets. Therefore, there is no guarantee for the order of the aliases. Do you think this should be an ordered collection? It didn't make much sense to me to preserve the order.

Reply:
Order doesn't need to be stable here IMO, we can improve this later using a `LinkedHashSet` if there is demand for it.

Reply:
Got it, thanks! As long as there are no wildcards in the aliases, the order doesn't seem to matter, indeed.